src/app/views/forgotten-password/setForgottenPasswordController.jsx

```diff
@@ -11,12 +15,115 @@ export class SetForgottenPasswordController extends Component {
+                });
+            });
+    }
+
+    handlePasswordResetError(error) {
+        const notification = {
+            type: "warning",
+            isDismissable: true,
+            autoDismiss: 15000
+        };
+
+        if (error != null && error.status != null) {
+            if (error.status === 400) {
+                // All validation errors will be captured by this; if using the web interface validation is checked before you can submit
+                console.error("Unable to validate the type, email, password, or verification_token in the request");
+                log.event("Unable to validate the type, email, password, or verification_token in the request", log.error(error));
+                notification.message = errCodes.RESET_PASSWORD_VALIDATION_ERR;
+            } else if (error.status === 500) {
+                console.error("Invalid request body");
+                log.event("Invalid request body", log.error(error));
+                notification.message = errCodes.RESET_PASSWORD_REQUEST_UNEXPECTED_ERR;
+            } else if (error.status === 501) {
+                console.error("Requested unimplemented password change type");
+                log.event("Requested unimplemented password change type", log.error(error));
+                notification.message = errCodes.RESET_PASSWORD_REQUEST_UNEXPECTED_ERR;
+            } else {
+                console.error(errCodes.RESET_PASSWORD_REQUEST_UNEXPECTED_ERR);
+                log.event(errCodes.RESET_PASSWORD_REQUEST_UNEXPECTED_ERR, log.error(error));
+                notification.message = errCodes.RESET_PASSWORD_REQUEST_UNEXPECTED_ERR;
+            }
+        } else {
+            console.error(errCodes.RESET_PASSWORD_REQUEST_UNEXPECTED_ERR);
+            log.event(errCodes.RESET_PASSWORD_REQUEST_UNEXPECTED_ERR, log.error(error));
+            notification.message = errCodes.RESET_PASSWORD_REQUEST_UNEXPECTED_ERR;
+        }
+        notifications.add(notification);
+    }
+
+    validityCheck(isValid, password) {
+        // Only show input error if user had previously tried to submit password and it is still invalid
+        const showInputError = !isValid && this.state.showInputError;
+        this.setState({
+            passwordIsValid: isValid,
+            password: password,
+            showInputError: showInputError
+        });
     }
 
     render() {
-        const screenToShow = this.state.hasSubmitted ? <SetForgottenPasswordConfirmed /> : <SetForgottenPasswordRequest />;
+        const setForgottenPasswordRequestProps = {
+            validityCheck: this.validityCheck,
+            isValid: this.state.passwordIsValid,
+            onSubmit: this.onSubmit,
+            heading: "Create a new password",
+            buttonText: "Confirm password",
+            showInputError: this.state.showInputError,
+            isSubmitting: this.state.isSubmitting
+        };
+
+        const screenToShow = this.state.hasSubmitted ? (
```

Comment:
i would probably simplify this a bit:
if you make changes as per my comment above: 
`if( this.state.status === STATUS. SUBMITTED) return  <SetForgottenPasswordConfirmed /> 
return <SetForgottenPasswordRequest {...setForgottenPasswordRequestProps} />` don't store this in constant, no need for it :-)